tests/integration/test_metric_collection.py

```diff
@@ -7,34 +7,236 @@
 import difflib
 
 endpoint = os.getenv("TIMESERIES_MOCK_ENDPOINT", "localhost:8088")
+ci = os.getenv("CI", "")
 url = f'http://{endpoint}/metrics.json'
 
-
 endpointPrometheus = os.getenv("PROMETHEUS_MOCK_ENDPOINT", "localhost:8080")
 urlMetrics = [f'http://{endpointPrometheus}/metrics',
               f'http://{endpointPrometheus}/federate?match%5B%5D=container_cpu_usage_seconds_total&match%5B%5D=container_spec_cpu_quota&match%5B%5D=container_spec_cpu_period&match%5B%5D=container_memory_working_set_bytes&match%5B%5D=container_spec_memory_limit_bytes&match%5B%5D=container_cpu_cfs_throttled_periods_total&match%5B%5D=container_cpu_cfs_periods_total&match%5B%5D=container_fs_reads_total&match%5B%5D=container_fs_writes_total&match%5B%5D=container_fs_reads_bytes_total&match%5B%5D=container_fs_writes_bytes_total&match%5B%5D=container_fs_usage_bytes&match%5B%5D=container_network_receive_bytes_total&match%5B%5D=container_network_transmit_bytes_total&match%5B%5D=container_network_receive_packets_total&match%5B%5D=container_network_transmit_packets_total&match%5B%5D=container_network_receive_packets_dropped_total&match%5B%5D=container_network_transmit_packets_dropped_total&match%5B%5D=apiserver_request_total&match%5B%5D=kubelet_volume_stats_available_percent&match%5B%5D=%7B__name__%3D%22kubernetes_build_info%22%2C+job%3D~%22.%2Aapiserver.%2A%22%7D']
 
-with open('expected_metric_names.txt', "r", newline='\n') as file_with_expected_metric_names:
-    expected_metric_names = file_with_expected_metric_names.read().splitlines()
 
 def test_expected_metric_names_are_generated():
-    retry_until_ok(url, assert_test_metric_names_found,
-                   print_failure_metric_names)
+    expected_metric_names = []
+
+    with open(os.path.join(os.path.dirname(__file__), 'expected_metric_names.txt'), "r", newline='\n') as file_with_expected_metric_names:
+        expected_metric_names = file_with_expected_metric_names.read().splitlines()
 
-# Commented till we resolve testing against real cluster
-# def test_expected_otel_message_content_is_generated():
-#     retry_until_ok(url, assert_test_expected_otel_message_content_is_generated,
-#                    print_failure_otel_content)
+    retry_until_ok(url, 
+                   lambda content: assert_metric_names_found(content, expected_metric_names),
+                   lambda content: print_failure_metric_names(content, expected_metric_names))
+    
+test_cases = [
```

Comment:
These would be nice if it's split into separate files? Otherwise PR looks fine to me.

Reply:
Ok I will split that i followup PR, i will merge this to avoid nerge conflicts